### ISSUE TITLE
1.2.0: test suite is failing

### ISSUE THREAD

**Issue by kloczek** [PackageKit/PackageKit#398]
```
+ cd PackageKit-1.2.0
+ /usr/bin/meson test -C x86_64-redhat-linux-gnu --num-processes 48 --print-errorlogs
ninja: Entering directory `/home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.0/x86_64-redhat-linux-gnu'
ninja: no work to do.
1/2 pk-test-private OK             0.05s
2/2 pk-self-test    FAIL           0.51s (killed by signal 5 SIGTRAP)

Ok:                 1
Expected Fail:      0
Fail:               1
Unexpected Pass:    0
Skipped:            0
Timeout:            0


The output from the failed tests:

2/2 pk-self-test    FAIL           0.51s (killed by signal 5 SIGTRAP)

--- command ---
13:32:43 PYTHONPATH='/home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.0/x86_64-redhat-linux-gnu/lib/python' /home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.0/x86_64-redhat-linux-gnu/src/pk-self-test
--- stdout ---
# random seed: R02Sf1b32800c560c341270f21693037989d
Bail out! FATAL-WARNING: you need to compile with -Dlocal_checkout=true for ninja test support
--- stderr ---

** (/home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.0/x86_64-redhat-linux-gnu/src/pk-self-test:1715193): WARNING **: 14:32:43.868: you need to compile with -Dlocal_checkout=true for ninja test support
-------

Full log written to /home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.0/x86_64-redhat-linux-gnu/meson-logs/testlog.txt
```

**Comment by kloczek** [comment]
After add missing option `-Dlocal_checkout=true` to meson aparams it still fails
```
+ /usr/bin/meson test -C x86_64-redhat-linux-gnu --num-processes 48 --print-errorlogs
ninja: Entering directory `/home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.0/x86_64-redhat-linux-gnu'
ninja: no work to do.
1/2 pk-test-private OK             0.07s
2/2 pk-self-test    FAIL           16.98s (killed by signal 5 SIGTRAP)

Ok:                 1
Expected Fail:      0
Fail:               1
Unexpected Pass:    0
Skipped:            0
Timeout:            0


The output from the failed tests:

2/2 pk-self-test    FAIL           16.98s (killed by signal 5 SIGTRAP)

--- command ---
13:48:40 PYTHONPATH='/home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.0/x86_64-redhat-linux-gnu/lib/python' /home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.0/x86_64-redhat-linux-gnu/src/pk-self-test
--- Listing only the last 100 lines from a long log. ---
# DEBUG: stdout 'percentage     90'
# DEBUG: stdout 'package        installed       gnome-power-manager;0.0.1;i386;data     More useless software'
# DEBUG: polling child_pid=1723985 (1/20)
# DEBUG: stdout 'percentage     100'
# DEBUG: the child exited with success
# DEBUG: emitting exit success
# DEBUG: spawn exit=0
# DEBUG: argv[0] '/home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.0/data/tests/pk-spawn-proxy.sh'
# DEBUG: envp[0] 'http_proxy=username:password@server:port'
# DEBUG: envp[1] 'ftp_proxy=username:password@server:port'
# DEBUG: creating new instance of /home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.0/data/tests/pk-spawn-proxy.sh
# GLib-DEBUG: posix_spawn avoided (fd close requested)
# DEBUG: stdout 'percentage     100'
# DEBUG: the child exited with success
# DEBUG: emitting exit success
# DEBUG: spawn exit=0
# DEBUG: argv[0] '/home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.0/data/tests/pk-spawn-test.sh'
# DEBUG: creating new instance of /home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.0/data/tests/pk-spawn-test.sh
# GLib-DEBUG: posix_spawn avoided (fd close requested)
# DEBUG: stdout 'percentage     0'
# DEBUG: stdout 'percentage     10'
# DEBUG: polling child_pid=1723999 (1/20)
# DEBUG: stdout 'percentage     20'
# DEBUG: stdout 'percentage     30'
# DEBUG: polling child_pid=1723999 (1/20)
# DEBUG: sending SIGQUIT 1723999
# DEBUG: stdout 'percentage     40'
# DEBUG: stdout 'package        available       polkit;0.0.1;i386;data  PolicyKit daemon'
# DEBUG: stdout 'package        installed       polkit-gnome;0.0.1;i386;data    PolicyKit helper (?) for GNOME'
# DEBUG: polling child_pid=1723999 (1/20)
# DEBUG: stdout 'package        available       ConsoleKit      System console checker'
# DEBUG: stdout 'percentage     50'
# DEBUG: polling child_pid=1723999 (1/20)
# DEBUG: stdout 'percentage     60'
# DEBUG: sending SIGKILL 1723999
# DEBUG: the child process was terminated by SIGKILL
# DEBUG: emitting exit sigkill
# DEBUG: spawn exit=5
# DEBUG: argv[0] '/home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.0/data/tests/pk-spawn-test.sh'
# DEBUG: creating new instance of /home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.0/data/tests/pk-spawn-test.sh
# GLib-DEBUG: posix_spawn avoided (fd close requested)
# DEBUG: stdout 'percentage     0'
# DEBUG: stdout 'percentage     10'
# DEBUG: polling child_pid=1724009 (1/20)
# DEBUG: stdout 'percentage     20'
# DEBUG: stdout 'percentage     30'
# DEBUG: polling child_pid=1724009 (1/20)
# DEBUG: sending SIGQUIT 1724009
# DEBUG: stdout 'percentage     40'
# DEBUG: stdout 'package        available       polkit;0.0.1;i386;data  PolicyKit daemon'
# DEBUG: stdout 'package        installed       polkit-gnome;0.0.1;i386;data    PolicyKit helper (?) for GNOME'
# DEBUG: polling child_pid=1724009 (1/20)
# DEBUG: stdout 'package        available       ConsoleKit      System console checker'
# DEBUG: stdout 'percentage     50'
# DEBUG: polling child_pid=1724009 (1/20)
# DEBUG: stdout 'percentage     60'
# DEBUG: stdout 'percentage     70'
# DEBUG: polling child_pid=1724009 (1/20)
# DEBUG: stdout 'percentage     80'
# DEBUG: stdout 'percentage     90'
# DEBUG: stdout 'package        installed       gnome-power-manager;0.0.1;i386;data     More useless software'
# DEBUG: polling child_pid=1724009 (1/20)
# DEBUG: stdout 'percentage     100'
# DEBUG: the child exited with success
# DEBUG: emitting exit sigquit
# DEBUG: spawn exit=4
# DEBUG: argv[0] '/home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.0/data/tests/pk-spawn-test-sigquit.py'
# DEBUG: creating new instance of /home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.0/data/tests/pk-spawn-test-sigquit.py
# GLib-DEBUG: posix_spawn avoided (fd close requested)
# DEBUG: stdout 'percentage     0'
# DEBUG: stdout 'percentage     10'
# DEBUG: stdout 'percentage     20'
# DEBUG: polling child_pid=1724025 (1/20)
# DEBUG: stdout 'percentage     30'
# DEBUG: sending SIGQUIT 1724025
# DEBUG: the child exited with success
# DEBUG: emitting exit sigquit
# DEBUG: spawn exit=4
# DEBUG: argv[0] '/home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.0/data/tests/pk-spawn-test-profiling.sh'
# DEBUG: creating new instance of /home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.0/data/tests/pk-spawn-test-profiling.sh
# GLib-DEBUG: posix_spawn avoided (fd close requested)
# DEBUG: killing as still running in finalize
# DEBUG: sending SIGQUIT 1724026
# DEBUG: argv[0] '/home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.0/data/tests/pk-spawn-dispatcher.py'
# DEBUG: argv[1] 'search-name'
# DEBUG: argv[2] 'none'
# DEBUG: argv[3] 'power manager'
# DEBUG: envp[0] 'NETWORK=TRUE'
# DEBUG: envp[1] 'LANG=C'
# DEBUG: envp[2] 'BACKGROUND=TRUE'
# DEBUG: envp[3] 'INTERACTIVE=TRUE'
# DEBUG: envp[4] 'UID=500'
# DEBUG: creating new instance of /home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.0/data/tests/pk-spawn-dispatcher.py
# GLib-DEBUG: posix_spawn avoided (fd close requested)
Bail out! FATAL-WARNING: the child exited with return code 1
--- stderr ---

** (/home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.0/x86_64-redhat-linux-gnu/src/pk-self-test:1723975): WARNING **: 14:48:56.866: the child exited with return code 1
```

**Comment by kloczek** [comment]
Looks like just released 1.2.3 still has that issue
```
+ /usr/bin/meson test -C x86_64-redhat-linux-gnu --num-processes 48 --print-errorlogs
ninja: Entering directory `/home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.3/x86_64-redhat-linux-gnu'
ninja: no work to do.
1/2 pk-test-private        OK               0.05s
2/2 pk-self-test           FAIL            15.95s   killed by signal 5 SIGTRAP
>>> PYTHONPATH=/home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.3/x86_64-redhat-linux-gnu/lib/python MALLOC_PERTURB_=203 /home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.3/x86_64-redhat-linux-gnu/src/pk-self-test
――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――― ✀  ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
Listing only the last 100 lines from a long log.
# DEBUG: stdout 'package        available       polkit;0.0.1;i386;data  PolicyKit daemon'
# DEBUG: stdout 'package        installed       polkit-gnome;0.0.1;i386;data    PolicyKit helper (?) for GNOME'
# DEBUG: polling child_pid=308284 (1/20)
# DEBUG: stdout 'package        available       ConsoleKit      System console checker'
# DEBUG: stdout 'percentage     50'
# DEBUG: polling child_pid=308284 (1/20)
# DEBUG: stdout 'percentage     60'
# DEBUG: stdout 'percentage     70'
# DEBUG: polling child_pid=308284 (1/20)
# DEBUG: stdout 'percentage     80'
# DEBUG: stdout 'percentage     90'
# DEBUG: stdout 'package        installed       gnome-power-manager;0.0.1;i386;data     More useless software'
# DEBUG: polling child_pid=308284 (1/20)
# DEBUG: stdout 'percentage     100'
# DEBUG: the child exited with success
# DEBUG: emitting exit success
# DEBUG: spawn exit=0
# DEBUG: argv[0] '/home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.3/data/tests/pk-spawn-proxy.sh'
# DEBUG: envp[0] 'http_proxy=username:password@server:port'
# DEBUG: envp[1] 'ftp_proxy=username:password@server:port'
# DEBUG: creating new instance of /home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.3/data/tests/pk-spawn-proxy.sh
# DEBUG: stdout 'percentage     100'
# DEBUG: the child exited with success
# DEBUG: emitting exit success
# DEBUG: spawn exit=0
# DEBUG: argv[0] '/home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.3/data/tests/pk-spawn-test.sh'
# DEBUG: creating new instance of /home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.3/data/tests/pk-spawn-test.sh
# DEBUG: stdout 'percentage     0'
# DEBUG: stdout 'percentage     10'
# DEBUG: polling child_pid=308301 (1/20)
# DEBUG: stdout 'percentage     20'
# DEBUG: sending SIGQUIT 308301
# DEBUG: stdout 'percentage     30'
# DEBUG: polling child_pid=308301 (1/20)
# DEBUG: stdout 'percentage     40'
# DEBUG: stdout 'package        available       polkit;0.0.1;i386;data  PolicyKit daemon'
# DEBUG: stdout 'package        installed       polkit-gnome;0.0.1;i386;data    PolicyKit helper (?) for GNOME'
# DEBUG: polling child_pid=308301 (1/20)
# DEBUG: stdout 'package        available       ConsoleKit      System console checker'
# DEBUG: stdout 'percentage     50'
# DEBUG: sending SIGKILL 308301
# DEBUG: the child process was terminated by SIGKILL
# DEBUG: emitting exit sigkill
# DEBUG: spawn exit=5
# DEBUG: argv[0] '/home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.3/data/tests/pk-spawn-test.sh'
# DEBUG: creating new instance of /home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.3/data/tests/pk-spawn-test.sh
# DEBUG: stdout 'percentage     0'
# DEBUG: stdout 'percentage     10'
# DEBUG: polling child_pid=308309 (1/20)
# DEBUG: stdout 'percentage     20'
# DEBUG: stdout 'percentage     30'
# DEBUG: polling child_pid=308309 (1/20)
# DEBUG: sending SIGQUIT 308309
# DEBUG: stdout 'percentage     40'
# DEBUG: stdout 'package        available       polkit;0.0.1;i386;data  PolicyKit daemon'
# DEBUG: stdout 'package        installed       polkit-gnome;0.0.1;i386;data    PolicyKit helper (?) for GNOME'
# DEBUG: polling child_pid=308309 (1/20)
# DEBUG: stdout 'package        available       ConsoleKit      System console checker'
# DEBUG: stdout 'percentage     50'
# DEBUG: polling child_pid=308309 (1/20)
# DEBUG: stdout 'percentage     60'
# DEBUG: stdout 'percentage     70'
# DEBUG: polling child_pid=308309 (1/20)
# DEBUG: stdout 'percentage     80'
# DEBUG: stdout 'percentage     90'
# DEBUG: stdout 'package        installed       gnome-power-manager;0.0.1;i386;data     More useless software'
# DEBUG: polling child_pid=308309 (1/20)
# DEBUG: stdout 'percentage     100'
# DEBUG: the child exited with success
# DEBUG: emitting exit sigquit
# DEBUG: spawn exit=4
# DEBUG: argv[0] '/home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.3/data/tests/pk-spawn-test-sigquit.py'
# DEBUG: creating new instance of /home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.3/data/tests/pk-spawn-test-sigquit.py
# DEBUG: stdout 'percentage     0'
# DEBUG: stdout 'percentage     10'
# DEBUG: polling child_pid=308322 (1/20)
# DEBUG: stdout 'percentage     20'
# DEBUG: stdout 'percentage     30'
# DEBUG: sending SIGQUIT 308322
# DEBUG: the child exited with success
# DEBUG: emitting exit sigquit
# DEBUG: spawn exit=4
# DEBUG: argv[0] '/home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.3/data/tests/pk-spawn-test-profiling.sh'
# DEBUG: creating new instance of /home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.3/data/tests/pk-spawn-test-profiling.sh
# DEBUG: killing as still running in finalize
# DEBUG: sending SIGQUIT 308323
# DEBUG: argv[0] '/home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.3/data/tests/pk-spawn-dispatcher.py'
# DEBUG: argv[1] 'search-name'
# DEBUG: argv[2] 'none'
# DEBUG: argv[3] 'power manager'
# DEBUG: envp[0] 'NETWORK=TRUE'
# DEBUG: envp[1] 'LANG=C'
# DEBUG: envp[2] 'BACKGROUND=TRUE'
# DEBUG: envp[3] 'INTERACTIVE=TRUE'
# DEBUG: envp[4] 'UID=500'
# DEBUG: creating new instance of /home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.3/data/tests/pk-spawn-dispatcher.py
Bail out! FATAL-WARNING: the child exited with return code 1
stderr:

** (/home/tkloczko/rpmbuild/BUILD/PackageKit-1.2.3/x86_64-redhat-linux-gnu/src/pk-self-test:308274): WARNING **: 16:39:36.844: the child exited with return code 1
――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――


Summary of Failures:

2/2 pk-self-test    FAIL            15.95s   killed by signal 5 SIGTRAP


Ok:                 1
Expected Fail:      0
Fail:               1
Unexpected Pass:    0
Skipped:            0
Timeout:            0
```

**Comment by kloczek** [comment]
Gentle ping .. any update? 🤔
Just tested 1.2.6 and looks like still is not resolved.
